pandas/io/excel/_base.py

```diff
@@ -1069,26 +1069,37 @@ def __init__(
 
             xlrd_version = LooseVersion(get_version(xlrd))
 
-        if xlrd_version is not None and isinstance(path_or_buffer, xlrd.Book):
-            ext = "xls"
-        else:
-            ext = inspect_excel_format(
-                content_or_path=path_or_buffer, storage_options=storage_options
-            )
-
+        ext = None
         if engine is None:
+            # Only determine ext if it is needed
+            if xlrd_version is not None and isinstance(path_or_buffer, xlrd.Book):
```

Comment:
could probably remove the nesting (I assume we always have to use `ext="xls"` even if `engine` is not `None`):

```py
if xlrd_version is not None and isinstance(path_or_buffer, xlrd.Book):
    ...
elif engine is not None:
    ....
```